crates/cargo-bsp/src/cargo_communication/request_actor.rs

```diff
@@ -898,4 +898,468 @@ pub mod compile_request_tests {
+
+            // tests started
+            assert_json_snapshot!(receiver_from_actor.recv().unwrap(),{
+                ".params.taskId.id" => RANDOM_TASK_ID,
+                ".params.eventTime" => TIMESTAMP,
+            }
+            ,@r###"
+            {
+              "method": "build/taskStart",
+              "params": {
+                "eventTime": "timestamp",
+                "message": "Started target testing",
+                "taskId": {
+                  "id": "random_task_id",
+                  "parents": [
+                    "test_origin_id"
+                  ]
+                }
+              }
+            }
+            "###);
+
+            drop(sender_to_actor);
+
+            // tests finished
+            assert_json_snapshot!(receiver_from_actor.recv().unwrap(),{
+                ".params.taskId.id" => RANDOM_TASK_ID,
+                ".params.eventTime" => TIMESTAMP,
+            }
+            ,@r###"
+            {
+              "method": "build/taskFinish",
+              "params": {
+                "eventTime": "timestamp",
+                "message": "Finished target testing",
+                "status": 2,
+                "taskId": {
+                  "id": "random_task_id",
+                  "parents": [
+                    "test_origin_id"
+                  ]
+                }
+              }
+            }
+            "###);
+
+            // main task finished
+            assert_json_snapshot!(receiver_from_actor.recv().unwrap(),{
+                ".params.eventTime" => TIMESTAMP,
+            }
+            ,@r###"
+            {
+              "method": "build/taskFinish",
+              "params": {
+                "eventTime": "timestamp",
+                "status": 2,
+                "taskId": {
+                  "id": "test_origin_id"
+                }
+              }
+            }
+            "###);
+
+            // response
+            assert_json_snapshot!(receiver_from_actor.recv().unwrap(), @r###"
+            {
+              "id": "test_req_id",
+              "result": {
+                "originId": "test_origin_id",
+                "statusCode": 2
+              }
+            }
+            "###);
+            no_more_msg(receiver_from_actor);
+        }
+
+        #[test]
+        fn suite_started() {
+            let (mut req_actor, receiver_from_actor, _cancel_sender) =
+                default_req_actor::<Test>(MockCargoHandler::new(), default_test_params());
+
+            let suite_started = SuiteStarted { test_count: 1 };
+
+            req_actor.handle_event(CargoEvent(CargoStdout(TextLine(
+                to_string(&TestType::Suite(SuiteEvent::Started(suite_started))).unwrap(),
+            ))));
+
+            assert_json_snapshot!(receiver_from_actor.recv().unwrap(), {
+                ".params.taskId.id" => RANDOM_TASK_ID,
+                ".params.taskId.parents" => format!("{RANDOM_TASK_ID}"),
```

Comment:
Skoro i tak nie tworzysz tutaj vectora, to po co jest ten format i czemu nie ma po prostu RANDOM_TASK_ID?